### DISCUSSION TITLE
Action for making a roborock vacuum a specific room configured in the official app?

### DISCUSSION THREAD

**Discussion by raine** [humbertogontijo/homeassistant-roborock#141] [Q&A] [answered]
I did some research and it looks like it might be possible with `vacuum.send_command` or `roborock.vacuum_clean_segment`.

But how does one obtain an id for the room?

What should the YAML for the action look like?

I'm attempting to set up an automation like so:

![image](https://user-images.githubusercontent.com/11027/212131944-08b0c287-c905-4f77-8f3b-e0e52bee3860.png)

Thanks.

**Comment by jdejaegh** [comment, marked as answer]
Not sure to understand correctly your question.  As I understood, you would like to have your Roborock clean a specific room (that you defined in the app) on the press of a button.  Am I correct?

To do so, you service call would look something like the following:
```yml
service: roborock.vacuum_clean_segment
data:
  segments: [20]
target:
  entity_id: vacuum.roborock_s7
```
In `data.segments` you can put multiple rooms if you wish, like `[20, 16, 1]`.

In order to find what are the correct number for you rooms, you might use [PiotrMachowski/lovelace-xiaomi-vacuum-map-card](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card).  

- Install the card from HACS
- Add the card to one of your dashboards
- Select the camera entity and the vacuum entity provided by the Roborock integration
- Click on "Generate static config" and "Generate room configs"
- Now you should have a card with the map from Roborock
- You can change the mode to room cleaning using the blue button on the left
- All the rooms are named "Room xx", where xx is the number of the room: this is the number you are looking for

**Reply by raine** [reply]
Thanks for the reply, Jules. You understood correctly.

I installed that card and don't see it appearing anywhere in the UI. I have one dashboard which is controlled by HA.

Surely there's a simpler way to get the room ids though? I wonder if this integration would reveal them somehow, perhaps through debug logs somewhere.

**Reply by raine** [reply]
Managed to figure it out the way you recommended.

Had to add a new dashboard and then that map card. Rest was pretty straightforward. It even had a way to copy the service call though it was in JSON format for some reason. But it had the correct parameters so it was simple to adapt to your example.

Cheers!

**Comment by smartmatic** [comment]
I‘ve installed the Xiaomi vacuum map card but i did not get a rooms config. Once i click Generate Rooms Config nothing happens means. Any help is appreciated!

**Reply by Lash-L** [reply]
> ve installed the Xiaomi vacuum map card but i did not get a rooms config. Once i click Generate Rooms Config nothing happens means. Any help is appreciated

You have to go to the yaml editor for it there's a button somewhere on there that says "show code editor"

**Reply by smartmatic** [reply]
> I‘ve installed the Xiaomi vacuum map card but i did not get a rooms config. Once i click Generate Rooms Config nothing happens means. Any help is appreciated!

I see the button, thats not the issue. When i click the button i got no rooms config added to the code or see any room numbers

![IMG_0343](https://user-images.githubusercontent.com/74932663/229695560-27d2039e-b9c3-4a2b-b27a-e379afa1e844.jpeg)


Are there any requirements that Generate room config is executed?

**Reply by dilorenzo1987** [reply]
Running into the same issue. Did anyone solve this?

**Reply by jata1** [reply]
I had the same issue but resolved this by creating a new dashboard page then add just the Xiaomi vacuum map card to this new dashboard. After doing this the buttons generated the data correctly for the loaded map.

Then you need to execute the service call (Vacuum camera map load multi map) for each floor/map - first map had id 0, second id 1 etc...

**Reply by bunjg** [reply]
I finally got the room numbers by changing to "Rooms" for the cleaning mode after setting up the card.  You need to make sure and hit "Generate Rooms Config" in order to get the "Rooms" dropdown option.
![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/48077320/bed8a8d6-9b2b-4cda-96f6-c82b2639807e)
